.teamcity/_self/projects/WebApp.kt

```diff
@@ -1201,3 +1202,81 @@ object QuarantinedE2ETests: E2EBuildType(
 	buildTriggers = {
 	}
 )
+
+object DataViewsChangelogCheck : BuildType({
+	id("calypso_WebApp_DataViews_Changelog_Check")
+	uuid = "c494fae3-e0e1-49e2-9219-ea59cc17c499"
+	name = "DataViews Changelog Check"
+	description = "Ensures that changes to the packages/dataviews fork are accompanied by changes to its changelog"
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	steps {
+		bashNodeScript {
+			name = "Check DataViews changelog"
+			scriptContent = """
+				#!/usr/bin/env bash
+				set -e
+
+				# Check if there are any changes to packages/dataviews
+				DATAVIEWS_CHANGES=${'$'}(git diff --name-only refs/remotes/origin/trunk...HEAD \
+					| grep -c ^packages/dataviews/ \
+					|| true)
+
+				if [ "${'$'}DATAVIEWS_CHANGES" -eq 0 ]; then
+					exit 0
+				fi
+
+				# Check if the changelog file was modified
+				CHANGELOG_MODIFIED=${'$'}(git diff --name-only refs/remotes/origin/trunk...HEAD \
+					| grep -c '^packages/dataviews/CHANGELOG.automattic.md${'$'}' \
+					|| true)
+
+				if [ "${'$'}CHANGELOG_MODIFIED" -eq 0 ]; then
```

Comment:
Doesn't `CHANGELOG_MODIFIED=0` mean there are no changes?

```suggestion
				if [ "${'$'}CHANGELOG_MODIFIED" -eq 1 ]; then
```

Reply:
Right, but isn't that what we want? The overall flow is:

```
if no changes in dataviews:
  return

if no changes in changelog (but there are changes in dataviews):
  throw
```

Btw, `CHANGELOG_MODIFIED` is a poor variable name, so I'll change it. It could be something like `CHANGELOG_MODIFICATIONS`. The point is that it's a count (`grep`'s `-c` option). The other way to do it -- now with fresh eyes -- is to ignore grep's output and look at exit codes:

```sh
if git diff ... | grep -q ^packages/dataviews/CHANGELOG.automattic.md; then
  echo ERROR
fi
```

Reply:
Oh, right. At this point in the flow, there will be changes to dataviews, but none to the changelog. Forgot to take my morning coffee before commenting.